app/models/course/achievement.rb

```diff
@@ -11,4 +11,13 @@ class Course::Achievement < ActiveRecord::Base
   def set_defaults
     self.weight ||= 10
   end
+
+  # Returns all achievements that this achievement may have as a condition.
+  #
+  # @return [Array<Course::Achievement>]
+  def valid_achievement_conditions
+    course.achievements - [self] -
+      specific_conditions.select { |c| c.is_a?(Course::Condition::Achievement) }.
+        map(&:achievement)
+  end
```

Comment:
Interim solution, just inlined the method I removed.

I get `ActiveRecord::EagerLoadPolymorphicError` when trying to join with `achievement.conditions.joins(:actable)`. Found [this](http://stackoverflow.com/a/16124295) but it involves declaring specialised associations, which seems to defeat the purpose of polymorphism?